File: golang/lib/execution/kurtosis_lambda_configurator.go
/*
 * Copyright (c) 2020 - present Kurtosis Technologies LLC.
 * All Rights Reserved.
 */

package execution

import (
	"github.com/kurtosis-tech/kurtosis-lambda-api-lib/golang/lib/kurtosis-lambda"
)

type KurtosisLambdaConfigurator interface {
	ParseParamsAndCreateKurtosisLambda(serializedCustomParamsStr string) (kurtosis_lambda.KurtosisLambda, error)
}

File: typescript/src/lib/execution/kurtosis_lambda_configurator.ts
import { Result } from "neverthrow";
import { KurtosisLambda } from "../kurtosis_lambda/kurtosis_lambda";

// Docs available at https://docs.kurtosistech.com/kurtosis-lambda-api-lib/lib-documentation
export interface KurtosisLambdaConfigurator {
    parseParamsAndCreateKurtosisLambda(serializedCustomParamsStr: string): Result<KurtosisLambda, Error>;
}
